site/en/reference/cli/concepts/tcp.md

```diff
@@ -0,0 +1,147 @@
+the endpoint for communication. Assigning the IPv6 address and port is also
+referred to as "naming the endpoint."
+
+### `listen` command
+
+After you initialize the TCP module, you can run a `tcp listen` command to
+use the example TCP listener to listen for incoming connections on the
+TCP endpoint specified with its IPv6 address and port number.
+
+If you want the example listener to stop listening for incoming TCP connections,
+you can issue the `tcp stoplistening` command. 
+
+### `connect` command
+
+A `tcp connect` command can be used to connect the example TCP endpoint to a
+peer TCP endpoint address. You can then issue a `tcp send` command to send a
+message to the peer.
+
+### `send` command
+
+A `tcp send` command sends a message using the example TCP endpoint to the
+destination endpoint that was speciifed in a preceding `tcp connect` command.
+
+### `benchmark` commands
+
+Once a TCP connection is established between two nodes, you can use the
+`benchmark` commands to send large amounts of data between the nodes to test
+network bandwidth and performance. The number of transmitted bytes in milliseconds
+as well as the TCP Goodput will be provided in the `benchmark` results.
+
+### `abort` command
+
+To immediately and unceremoniously end a TCP connection, run the `tcp abort`
+command on either node to transition the TCP endpoint to a closed state.
+
+### `sendend` command
+
+When one node is done sending data to the other node, the first node can
+issue a `tcp sendend` command to alert the second node to no longer expect
+data. The second node can also send a `tcp sendend` to the first node.
+Once each node receives a `TCP: Disconnected` message, the TCP connection
+between the two nodes is torn down. It is recommended but not required to
+issue this command when data transfer is complete.
+
+## Send a message between two nodes
+
+1. On node 1, initialize the TCP CLI module, then listen for incoming connections
+   using the example TCP listener.
+
+   ```
+   > tcp init
+   > tcp listen :: 30000
+   ```
+
+   The use of `::` denotes that the `listen` should use the unspecified IPv6 address,
+   thereby having the TCP/IPv6 stack assign the IPv6 address. The port is 30000.  
+
+1. On Node 2, initialize the TCP CLI module, connect to node 1, and then send a
+   simple message. 
+
+   ```
+   > tcp init
+   > tcp connect fe80:0:0:0:a8df:580a:860:ffa4 30000
+   > tcp send hello
+   ```
+### Verification
+
+Based on the example steps shown above, the following output would be expected:
+
+1. After Node 2 runs the `tcp connect` command, Node 2 should receive:
+   the message `TCP: Connection established`.
+1. Node 1 should then receive the messages (with example IPv6 address and port):
+    * `Accepted connection from [fe80:0:0:0:8f3:f602:bf9b:52f2]:49152`
+    * `TCP: Connection established`
+1. After Node 2 runs the `tcp send` command, Node 1 should receive:
+   the message: `TCP: Received 5 bytes: hello` 
```

Comment:
These bullets don't have consistent formatting...

probably don't need the colon after "receive" in 1 and 3, then there's a colon after message in 3